pkg/tests/service/service.go

```diff
@@ -253,7 +256,7 @@ type testCluster struct {
 }
 
 // NewCluster construct a cluster for integration test.
-func NewCluster(t *testing.T, opt Options) (Cluster, error) {
+func NewCluster(t *testing.T, opt Options, taskService taskservice.TaskService) (Cluster, error) {
```

Comment:
Instead of passing in taskservice directly, add the taskstore option to the option and use the option's taskstore to initialize taskservice inside the cluster. In the test, a common taskstore instance per serviceg is fine. If Option is not passed in, the default memtaskstore is used